classes/monograph/PublishedMonographDAO.inc.php

```diff
@@ -381,6 +278,132 @@ function getSortDirection($sortOption) {
+		$publishedMonograph = parent::_fromRow($row);
+
+		// Add the additional PublishedMonograph data
+		$publishedMonograph->setDatePublished($this->datetimeFromDB($row['date_published']));
+		$publishedMonograph->setAudience($row['audience']);
+		$publishedMonograph->setAudienceRangeQualifier($row['audience_range_qualifier']);
+		$publishedMonograph->setAudienceRangeFrom($row['audience_range_from']);
+		$publishedMonograph->setAudienceRangeTo($row['audience_range_to']);
+		$publishedMonograph->setAudienceRangeExact($row['audience_range_exact']);
+		$publishedMonograph->setCoverImage(unserialize($row['cover_image']));
+
+		HookRegistry::call('PublishedMonographDAO::_fromRow', array(&$publishedMonograph, &$row));
+		return $publishedMonograph;
+	}
+
+
+	//
+	// Private helper methods.
+	//
+	/**
+	 * Retrieve all published monographs by associated object.
+	 * @param $pressId int The monograhps press id.
+	 * @param $assocType int optional The associated object type.
+	 * @param $assocId int optional The associated object id.
+	 * @param $searchText string optional Search text for title and authors.
+	 * @param $rangeInfo DBResultRange optional Object with result range information.
+	 * @param $sortBy int optional Sort monographs by passed column option.
+	 * @param $sortDirection int optional Sort monographs by passed direction.
+	 * @param $featuredOnly boolean optional Whether the monographs are featured on passed associated object or not.
+	 * @param $newReleasedOnly boolean optional Whether the monographs are marked as new releases on associated object or not.
+	 * @return DAOResultFactory DB Object that fetches monographs objects.
+	 */
+	private function _getByAssoc($pressId, $assocType = null, $assocId = null, $searchText = null, $rangeInfo = null, $sortBy, $sortDirection, $featuredOnly, $newReleasedOnly) {
+		// If no sort by options passed, sort by the passed associated object default.
+		if (!$sortBy || !$sortDirection) {
+			$sortOption = null;
+			switch ($assocType) {
+				case ASSOC_TYPE_PRESS:
+					$press = DAORegistry::getDAO('PressDAO')->getById($assocId);
+					$sortOption = $press->getSetting('catalogSortOption') ? $press->getSetting('catalogSortOption') : $this->getDefaultSortOption();
+					break;
+				case ASSOC_TYPE_SERIES:
+					$series = DAORegistry::getDAO('SeriesDAO')->getById($assocId);
+					$sortOption = $series->getSortOption() ? $series->getSortOption() : $this->getDefaultSortOption();
+					break;
+				case ASSOC_TYPE_CATEGORY:
+					$category = DAORegistry::getDAO('CategoryDAO')->getById($assocId);
+					$sortOption = $category->getSortOption() ? $category->getSortOption() : $this->getDefaultSortOption();
+					break;
+			}
+			$sortBy = $this->getSortBy($sortOption);
+			$sortDirection = $this->getSortDirection($sortOption);
+		}
+
+		// If no associated object is passed, we will get all monographs, featured/new released to the press or not.
+		$assocType = (int) $assocType?$assocType:ASSOC_TYPE_PRESS;
```

Comment:
Watch out -- the `(int)` cast gets applied to `$assocType`, not the result of the ternary statement.